module/VuFind/src/VuFind/Hierarchy/TreeRenderer/JSTree.php

```diff
@@ -205,7 +215,13 @@ protected function getContextualUrl($node, $context)
             return $this->getUrlFromRouteCache('collection', $node->id)
                 . '#tabnav';
         } else {
-            $url = $this->getUrlFromRouteCache($node->type, $node->id);
+            $type = $node->type;
+            if ('collection' === $type
+                && empty($this->config->Collections->collections)
+            ) {
+                $type = 'record';
+            }
+            $url = $this->getUrlFromRouteCache($type, $node->id);
             return $node->type == 'collection'
```

Comment:
Is it intentional that you are continuing to use `$node->type` instead of `$type` here? Looks to me like that is probably not the desired behavior, but maybe I'm overlooking something.